### PR TITLE
Xenial porting

### PR DESCRIPTION
**Pull request by UniversalSuperBox**
This PR adds our Xenial porting documentation and turns on TODOs. 

I'd suggest squashing it, my commit history is messy. Sorry...

### CODE REVIEW

File: porting/installing-16-04.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+The script will copy and extract the files to their proper places, then allow you to set the phablet user's password. If it gets all the way to ``rebooting device`` and doesn't seem to produce any errors, it's time to continue to the next step. If something goes wrong, please get in touch with us. If your device doesn't reboot automatically, reboot it using your recovery's interface.
+
+
+Get SSH access
+--------------
+
+When your device boots, it will likely stay at the bootloader screen. However, you should also get a new network connection on the computer you have it plugged in to. We will use this to debug the system.
+
+To confirm that your device has booted correctly, run ``dmesg -w`` and watch for "GNU/Linux device" in the output. If you instead get something similar to "ubports initrd i hit a nail", please get in contact with us so we can find out why.
```
Comment (vanyasem) [inline porting/installing-16-04.rst]: adding `watch ip link` would be also nice

File: porting/building-ubports-boot.rst [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+The initramfs of the ubports-boot image takes a commandline parameter, ``datapart=``, which points it to the block device that is normally mounted at ``/data`` in Android. This parameter is automatically embedded into the image at build time by finding ``/data`` in your device's default fstab and using its source. Unfortunately, this source is generally a node in ``/dev/block/bootdevice/by-name`` or something else strange, which is only available in an Android initramfs. To fix this, Halium uses the fixup-mountpoints script, which you are probably familiar with by now.
+
+The ubports-boot initramfs does not use the fixup-mountpoints script, nor does it put block devices in the proper place for mounting without modification. For this reason, we are going to have to edit your device's fstab.
+
+The first step to this process is figuring out where your fstab actually is. For most, this is inside ``$BUILDDIR/device/[manufacturer]/[codename]/rootdir/etc`` and it is named either ``fstab.qcom`` or ``fstab.devicename``. Open the file for editing.
```
Comment (doniks) [inline porting/building-ubports-boot.rst]: Let's not use too many different styles _variables_ in the instructions. I'd suggest ALLCAPS, but any consistent style is fine. Here you use three different ones:

- with a dollar sign $BUILDDIR
- with rectangular brackets [manufacturer]
- without any emphasis in fstab.devicename

Also, I think the appendix to fstab is the same as the foldername, so in conclusion I suggest: 

this is inside ``BUILDDIR/device/MANUFACTURER/CODENAME/rootdir/etc`` and it is named either ``fstab.qcom`` or ``fstab.CODENAME``.

File: porting/building-ubports-boot.rst [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+
+Edit kernel config
+------------------
+
+Ubuntu Touch requires a slightly different kernel config than Halium, including enabling Apparmor. Luckily, we have a nice script for this purpose, ``check-kernel-config``. It's included in the ubports-boot repository. Simply run it on your config as follows::
+
+    check-kernel-config path/to/my/defconfig -w
+
+You may to do this twice, it will likely fix things both times. Then, run the script without the ``-w`` flag to see if there are any more errors. If there are, fix them manually. Once finished, run the script without the ``-w`` flag one more time to make sure everything is correct.
```
Comment (doniks) [inline porting/building-ubports-boot.rst]: You may **have** to do this twice. **It** will

File: porting/installing-16-04.rst [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+Now that you've :doc:`built ubports-boot <building-ubports-boot>`, we'll use a script called ``rootstock-touch-install`` to install an Ubuntu Touch rootfs on your device.
+
+In order to install Ubuntu Touch, you will need a recovery with Busybox, such as TWRP, installed on your phone. You will also need ensure the /data partition is formatted with ext4 and does not have any encryption on it.
+
+
+Choose a rootfs
+---------------
+
+Before installing a rootfs, you'll need one to install. Which one you choose will depend on whether your device uses CAF sources or not. To check if your device contains any CAF source (it probably does), do a ``grep -r [term]`` for all of the following terms in $BUILDDIR/device/[manufacturer]/[codename]. If any search returns hits, you have a CAF device::
```
Comment (doniks) [inline porting/installing-16-04.rst]: Again, just **one** style, like 

BUILDDIR/device/MANUFACTURER/CODENAME

**Pull request merged**